### ISSUE TITLE
[ECOSYSTEM CHALLENGE] Using Remix and MetaMask on Moonbeam

### ISSUE THREAD

**Issue by albertov19** [0101coding/Gitcoin-HelloWorld#1]
Hi 0101coding,

Thanks for your submission. Unfortunately, you have not set the allowance (through the approve method) to the address specified in the instructions. Please when doing so let me know so I can check again for your submission.

Regards,
Alberto

**Comment by 0101coding** [comment]
Hi Alberto,

I set the allowance for the address in the constructor to 5 ether, Does that not suffice?

**Comment by albertov19** [comment]
Yes that should be enough. But I'm afraid I'm getting the same problem, are you sure you've set the correct address for the allowance? Remember you need to use the approve method, the allowance method is just to check it.

**Comment by 0101coding** [comment]
See snippet from the contract

` address public specialAddyToken = 0x12Cb274aAD8251C875c0bf6872b67d9983E53fDd; 
   constructor () public { 
       allowance_[msg.sender][specialAddyToken] = 5 * 1 ether;
       balanceOf_[msg.sender] = 1000 * 1 ether;  
   }
    mapping (address => uint256) balanceOf_;
    mapping (address => mapping(address => uint256)) allowance_;
    
    function allowance(address tokenOwner, address spender) override public view returns (uint remaining){
        remaining = allowance_[tokenOwner][spender];
    }
    function approve(address spender, uint256 tokens) override public  returns (bool success){
        allowance_[msg.sender][spender] = tokens;
        emit Approval(msg.sender, spender, tokens);
        success = true;
    }`

**Comment by 0101coding** [comment]
Hi albertov19
Check this newly deployed contract - 0x72fA4143Bd7629D725e1ce341077AaA241D8C974

**Comment by albertov19** [comment]
It is still not working :(. I recommend you use a normal ERC20 smart contract and then interact with it instead of trying to use methods in the constructor function. In any case I think the correct way of calling methods in the constructor function:

```
...
approve(0x12Cb274aAD8251C875c0bf6872b67d9983E53fDd, 5 ether);
...
```

But this really depends on the ERC20 token contract you are using

Regards
Alberto

**Comment by 0101coding** [comment]
Thanks very much albertov19, I will do as suggested

**Comment by 0101coding** [comment]
Thanks very much, albertov19. I think I found the issue in transferFrom function 


Regards

**Comment by 0101coding** [comment]
HI albertov19,

I think I fixed it. See new contract address - 0x9222757a3C4437Ea1664ee35bBc017F1207C0aeB

**Comment by albertov19** [comment]
Hey 0101coding,

You need to still do the correct allowance with the approve method in remix.

Contact me in Discord and I'll give you a hand in there.

Regards
Alberto
